File: app/src/main/java/com/funny/translation/fragements/CodeEditorViewModel.java
package com.funny.translation.fragements;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class CodeEditorViewModel extends ViewModel {
    // TODO: Implement the ViewModel
    MutableLiveData<String> code = new MutableLiveData<>("");
    MutableLiveData<String[]> keywords = new MutableLiveData<>(new String[]{});

    public MutableLiveData<String> getCode() {
        return code;
    }

    public void setCode(String code) {
        getCode().setValue(code);
    }

    public void setKeywords(String[] keywords) {
        getKeywords().setValue(keywords);
    }

    public MutableLiveData<String[]> getKeywords() {
        return keywords;
    }
}

File: codeeditor/src/main/java/com/funny/translation/codeeditor/vm/CodeEditorViewModel.kt
package com.funny.translation.codeeditor.vm

import android.app.Application
import androidx.lifecycle.MutableLiveData
import com.funny.translation.codeeditor.base.BaseViewModel
import com.funny.translation.codeeditor.extensions.readAssets

class CodeEditorViewModel(application: Application) : BaseViewModel(application) {
    lateinit var code : MutableLiveData<String>
    val displaySymbols = arrayOf("->","=", "{", "}", "(", ")", ",", ".", ";", "\"", "?", "+", "-", "*", "/")
    val insertSymbols = arrayOf("\t", "=","{}", "}", "(", ")", ",", ".", ";", "\"", "?", "+", "-", "*", "/")

    fun setCode(newCode:String){
        code.value = newCode
    }

    init {
        execute {
            code = MutableLiveData<String>(context.readAssets("js_template.js"))
        }
    }

}